src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx

```diff
@@ -1,5 +1,236 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { selectProviderSelectionInfo } from '../../../utils/selectors';
+import ResidentialAddress from '../../../components/ResidentialAddress';
+import { connect } from 'react-redux';
+import * as actions from '../../redux/actions';
+import { FETCH_STATUS } from '../../../utils/constants';
+import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
+import { distanceBetween } from '../../../utils/address';
+import { scrollAndFocus } from '../../../utils/scrollAndFocus';
+import ErrorMessage from '../../../components/ErrorMessage';
 
-export default function ProviderSelectionField() {
-  return <strong>List stuff goes here</strong>;
+const INITIAL_PROVIDER_DISPLAY_COUNT = 5;
+
+function ProviderSelectionField({
+  address,
+  formData,
+  onChange,
+  idSchema,
+  requestStatus,
+  requestProvidersList,
+  communityCareProviderList,
+}) {
+  const [checkedProvider, setCheckedProvider] = useState();
+  const [mounted, setMounted] = useState(false);
+  const [showProvidersList, setShowProvidersList] = useState(false);
+  const [providersListLength, setProvidersListLength] = useState(
+    INITIAL_PROVIDER_DISPLAY_COUNT,
+  );
+  const currentlyShownProvidersList = communityCareProviderList?.slice(
+    0,
+    providersListLength,
+  );
+  const loadingProviders =
+    requestStatus === FETCH_STATUS.loading ||
+    requestStatus === FETCH_STATUS.notStarted;
+  const providerSelected = 'id' in formData;
+  useEffect(() => {
+    requestProvidersList(address);
+    setMounted(true);
+  }, []);
+
+  useEffect(
+    () => {
+      if (showProvidersList && !loadingProviders) {
+        scrollAndFocus('h2');
+      } else if (mounted && !loadingProviders) {
+        scrollAndFocus('.va-button-link');
+      }
+    },
+    [showProvidersList],
+  );
+
+  return (
+    <div className="vads-u-background-color--gray-lightest small-screen:vads-u-padding--2 medium-screen:vads-u-padding--3">
+      {!showProvidersList &&
+        !providerSelected && (
+          <button
+            className="va-button-link"
+            type="button"
+            onClick={() => setShowProvidersList(true)}
+          >
+            <i className="fas fa-plus vads-u-padding-right--0p5" />
+            Choose a provider
+          </button>
+        )}
+      {!showProvidersList &&
+        providerSelected && (
+          <>
+            <span className="vads-u-display--block vads-u-font-weight--bold">
+              {formData?.name}
```

Comment:
I don't think you need the `?` after `formData`, since `providerSelected` can't be true if `formData` is falsy.

Reply:
Agreed, fixed.